ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -148,55 +149,92 @@ struct Assemble<'a> {
     ///
     /// Each assembly offset can have zero or more comment lines.
     comments: Cell<IndexMap<usize, Vec<String>>>,
+    /// Stack pointer offset from the base pointer of the interpreter frame including the
+    /// interpreter frame itself and all parent traces. Since all traces execute in the interpreter
+    /// frame, this trace adds to this value, making extra space on the stack. This then forms the
+    /// new `sp_offset` for any side-traces of this trace.
+    sp_offset: usize,
```

Comment:
Should we make this a new type? Then we would always know what it's doing.

Reply:
I'm unsure it's worth the extra hassle of having to convert between types every time we use it.

Reply:
In general, I think one can overuse newtypes, but this might be one case where it's worth doing. But, fair enough, we can do it in another PR.